### PR TITLE
feat: network switch

### PR DESCRIPTION
**Pull request by dzbo**
### Ticket ID

[DEV-8289](https://app.clickup.com/t/2645698/DEV-8289)

### Description

- handle network switch from ext

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-8771 Improve wallet loading performance -part I](https://app.clickup.com/t/2645698/DEV-8771)

**Comment by Hugoo** [comment]
Task linked: [DEV-8289 As a User, I want to switch from testnet to mainnet](https://app.clickup.com/t/2645698/DEV-8289)

**Review by richtera — APPROVED** [review]

File: composables/useBrowserExtension.ts [hunk 1]
```diff
@@ -100,18 +102,28 @@ const handleAccountsChanged = async (accounts: string[]) => {
...
+
 const handleDisconnect = () => {
   location.reload()
 }
 
-const addProviderEvents = async (provider: any) => {
+const addProviderEvents = async (provider: ProviderAPI) => {
   provider?.on?.('accountsChanged', handleAccountsChanged)
   provider?.on?.('disconnect', handleDisconnect)
+  provider?.on?.('chainChanged', handleChainChanged)
```
Comment (doubleppereira) [inline composables/useBrowserExtension.ts:120]: strange. I did this locally and it was working all. did it require any additional setup?

**Review by franckwei — APPROVED** [review]

**Comment by richtera** [comment]
@doubleppereira Could it be that the accountsChanges and chainChanged are triggered in such way that when chainChanged triggers there is no current value of accounts? Then it wouldn't show as being fired because while account is empty we don't attach the callbacks.

**Comment by doubleppereira** [comment]
> @doubleppereira Could it be that the accountsChanges and chainChanged are triggered in such way that when chainChanged triggers there is no current value of accounts? Then it wouldn't show as being fired because while account is empty we don't attach the callbacks.

yes it's possible to be that

**Pull request merged**